File: 557-reverse-words-in-a-string-iii/557-reverse-words-in-a-string-iii.js
/**
 * @param {string} s
 * @return {string}
 */
var reverseWords = function(s) {
    let word = [];
    let arr = [];
    for (let i = s.length; i > -1; i--){
        if (s[i] === " "){
            arr.push(word.join(''));
            word = [];
        } else {
            word.push(s[i]);
        }
    }
    arr.push(word.join(''));
    return arr.reverse().join(" ");

};

File: 557-reverse-words-in-a-string-iii/557-reverse-words-in-a-string-iii.py
class Solution:
    def reverseWords(self, s: str) -> str:
        word = []
        result = []
        for i in range(len(s) - 1, -1, -1):
            if s[i] == " ":
                result.append("".join(word))
                word = []
            else:
                word.append(s[i])
        result.append("".join(word))
        result.reverse()
        result = " ".join(result)
        return result